### PR TITLE
add coarse grid

### LINKED ISSUE #479: add coarse grid for laptop use
A grid with the coarse spatial and spectral resolution is required to be added to ww3_ufs1.1 test.

### PR DESCRIPTION
**Pull request by aliabdolali**
# Pull Request Summary
This PR adds a coarse grid in ww3_ufs1.1 to provide the option to the users to run this test on small PCs. 

## Description
The new grid has 5 degree resolution with the same inputs as ww3_ufs1.1 with 1 degree resolution with a coarser spectral resolution (12 directional and 18 spectral bins).
The option "ufscoarse" is added to matrix.base, if activated, the 5-degree grid is replaced by the 5 degree grid.
These tests can be run on small computers (laptops) in less than 5 sec.
Is a change of answers expected from this PR? No except ww3_ufs1.1 test.

### Issue(s) addressed
* Is there an issue associated with this development (bug fix, enhancement, new feature)?   #479 

- fixes #479

### Check list  
* Is your feature branch up to date with the authoritative repository (NOAA/develop)? Yes
* Make sure you have checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop) and [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number)
* Please list appropriate labels code managers should add for this PR:   
 _bug_, _documentation_, _enhancement_, _new feature_, ..
 
* Reviewers: @MatthewMasarik-NOAA @JessicaMeixner-NOAA 

### Commit Message
* add coarse grid option to ww3_ufs1.1 test

### Testing
* How were these changes tested? On Hera 
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Yes
* If a new feature was added, was a new regression test added?
* Have regression tests been run?
* Which compiler / HPC you used to run the regression tests in the PR? intel, impi and netcdf
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):    
Please indicate the expected changes in the outputs ([excluding the known list of non-identical tests](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-master#4-look-at-results)).
No changes except pre-known cases and ww3_ufs1.1 test.
* Please list which labels code managers should add to indicate code changes:    
_mod_def change_, _out_grd change_, _out_pnt change_, _restart file change_, _Regression test_, ...

### CODE REVIEW

File: regtests/ww3_ufs1.1/info [hunk 1]
```diff
@@ -8,7 +8,10 @@
 # hourly  wind speed (gfs_wind.nc) and current fields with 3 hrs interval   #
 # (current.nc).                                                             #
 # POINT: points. The unified point output grid is glo_1deg                  #
-#                                                                           #
+# --------------------------------------------------------------------------#
+# if "ufscoarse" option is selected, a coarser grid with 5 deg. resolution  #
```
Comment (JessicaMeixner-NOAA) [inline regtests/ww3_ufs1.1/info]: Is this referring to the "ufscoarse" option in matrix.base?

Comment (JessicaMeixner-NOAA) [reply regtests/ww3_ufs1.1/info]: Should this instead be expanded to explain grdset_c ?

File: regtests/bin/matrix_ncep [hunk 2]
```diff
@@ -180,44 +180,46 @@
...
+  export    multi03='y' # mww3_test_03 (three high and three low res grids).
+  export    multi04='y' # mww3_test_04 (swell on sea mount and/or current)
+  export    multi05='y' # mww3_test_05 (three-grid moving hurricane)
+  export    multi06='y' # mww3_test_06 (curvilinear grid tests)
+  export    multi07='y' # mww3_test_07 (unstructured grid tests)
+  export    multi08='y' # mww3_test_08 (wind and ice tests)
+  export    multi09='y' # mww3_test_09 (SMC multi grid test)
+
+  export        ufs='y' # The Unified Forecast System
+  export  ufscoarse='y' # Option for small PCs
```
Comment (JessicaMeixner-NOAA) [inline regtests/bin/matrix_ncep:219]: This option needs to be added to all matrix_* files and not just matrix_ncep.

Comment (aliabdolali) [reply regtests/bin/matrix_ncep:219]: done

File: regtests/ww3_ufs1.1/info (hunk 1 above)
Comment (aliabdolali) [reply regtests/ww3_ufs1.1/info]: added

File: regtests/bin/matrix.base [hunk 3]
```diff
@@ -2124,25 +2124,50 @@
...
+       echo "$rtst -w work_c -m grdset_c -f -p $mpi -n $np  -o all $ww3 ww3_ufs1.1"                           >> matrix.body
+        if [ "$npl_b4b" = 'y' ]
+        then
+           halfnp=$(($np / 2))
+           echo "$rtst -w work_c_npl -m grdset_c -f -p $mpi -n $halfnp -o all $ww3 ww3_ufs1.1"                >> matrix.body
+           echo "./bin/test.comp ww3_ufs1.1 work_c work_c_npl"                                                >> matrix.body
+        fi
+        if [ "$nth_b4b" = 'y' ]
+        then
+           echo "$rtst -w work_c_nth -m grdset_c -f -p $mpi -n $nr -t $nth -o all $ww3 ww3_ufs1.1"            >> matrix.body
```
Comment (JessicaMeixner-NOAA) [inline regtests/bin/matrix.base]: Need a new switch for this, the existing switch does not have OMPG and OMGH flags (as they are in mode/esmf/switch).  Not sure if this is the case for the other ufs tests as well or not.

Comment (aliabdolali) [reply regtests/bin/matrix.base]: I updated the switch for all ww3_ufs* cases and tested them. I do not know when I removed them. Thanks for catching it.

**Comment by aliabdolali** [comment]
The tests ran successfully on Hera **with pre-known non-b4b cases:** 

```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UQ_MPI_d2                     (7 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (8 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (8 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (10 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (6 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (8 files differ)
mww3_test_07/./work_PR3_UQ                     (3 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (6 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
```
**changes in ww3_ufs1.1 are expected as the test is changed from 1 deg to 5 degree:**
```
ww3_ufs1.1/./work_d                     (directory not found)
ww3_ufs1.1/./work_c_npl                     (directory not found)
ww3_ufs1.1/./work_c                     (directory not found)
ww3_ufs1.1/./work_c_nth                     (directory not found)
```
**The following are identical except the switch in the header of netcdf files:**
```
ww3_ufs1.2/./work_b                     (4 files differ)
ww3_ufs1.2/./work_a                     (4 files differ)
ww3_ufs1.3/./work_a                     (4 files differ)
```


[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/7238384/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/7238385/matrixCompSummary.txt)
[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/7238386/matrixDiff.txt)

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Pull request merged**